### ISSUE TITLE
[HOLD for payment 2023-08-02] LHN - If the first comment in a thread is deleted `[Deleted message]` appears in the LHN subtitle of the parent report

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23401]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any chat, say with user X
2. Send "one", "two", "three" as separate messages 
3. Thread off of "one"
4. Send a message in the thread
5. Delete "one" while inside of the thread
6. Look at the chat with user X on the LHN
7. The subtitle should say "three" but it says [message deleted] instead

## Expected Result:
The subtitle should say "three"

## Actual Result:
The subtitle should says `[Deleted message]`

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.43-7
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/6635c3e0-370c-4c56-80bd-9877ffb568ef

https://github.com/Expensify/App/assets/93399543/46b535f1-9fc7-4170-ab69-d6398b983a0a

https://github.com/Expensify/App/assets/93399543/426d14c7-3446-40c6-8826-c82f3f7a33fe

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689947132354429

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013902117a4d0a80c3</li>
        <li>Upwork Job ID: 1683594961925722112</li>
        <li>Last Price Increase: 2023-07-24</li>
    </ul>
</details>

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

`[Deleted message]` appears in LHN even after the last message is different

### What is the root cause of that problem?

This issue is from the [PR](https://github.com/Expensify/App/pull/22864)

If we delete the first message with child report, `isLastMessageDeletedParentAction` is set in the report.
https://github.com/Expensify/App/blob/48edd314cc730a5c43143ae560fa1643c17456cb/src/libs/actions/Report.js#L904
We set this value to null only when we add new comment. This is incorrect and is the root cause of the issue

The root cause of https://github.com/Expensify/App/issues/23450 is the same

### What changes do you think we should make in order to solve the problem?

Actually, we don't need the `isLastMessageDeletedParentAction` property in report. And some of the changes of the backend mentioned https://github.com/Expensify/App/issues/19722#issuecomment-1629765984 are not needed.

All we have to do:
- Include deleted parent action that has `childVisibleActionCount` > 0 in the visible actions
- If last visible action is a deleted parent action, we show [Deleted message]

Doing this would fix this issue and https://github.com/Expensify/App/issues/19722. No needs to change backend.

**Details**
1. Define a function to check if an anction is a visible deleted parent action(`childVisibleActionCount` > 0) in ReportActionUtils.js
```
function isVisibleDeletedParentAction(reportAction) {
    return lodashGet(reportAction, ['message', 0, 'isDeletedParentAction'], false) && lodashGet(reportAction, 'childVisibleActionCount', 0) > 0
}
```

2. In `getLastVisibleAction`, use `shouldReportActionBeVisible` instead of `!isDeletedAction`
3. In `getLastVisibleMessage`, add the following code
```
    if (isVisibleDeletedParentAction(action)) {
        return {
            lastMessageTranslationKey: 'parentReportAction.deletedMessage'
        }
    }
```

4. In `getLastMessageTextForReport` function in OptionsListUtils.js, get last action by `getLastVisibleAction` function
```
    const lastReportAction = ReportActionUtils.getLastVisibleAction(report.reportID);
```

and add the following 
```
    } else if (ReportActionUtils.isVisibleDeletedParentAction(lastReportAction)) {
        lastMessageTextFromReport = `${Localize.translateLocal(report.lastMessageTranslationKey || 'parentReportAction.deletedMessage')}`;
```


### What alternative solutions did you explore? (Optional)

**Comment by cead22** [comment]
> ## Expected Result:
> Should show last message in LHN

@kbecciv to clarify, do you mean the UI should be like
```
Test
Test
```
Instead of 
```
[Deleted message]
Test
```


I actually thought this is how this was designed to work, which is similar to slack. Was this discussed anywhere? I'm just curious since I'm just catching up on this issue, and my understanding was that this is how it's supposed to work

**Comment by marcaaron** [comment]
FWIW, I am doing some testing and the behavior is the same on Production and Staging.

**Comment by grgia** [comment]
I think there's some confusion here

There's two reports in this scenario, the parent report and the thread (child) report 

- the parent report should have the old title and the last message sent as subtitle
- the (child) report title should now be [Deleted message] and have the last message sent as subtitle

**Comment by marcaaron** [comment]
I am also not seeing the `[Deleted message]` show up for the message text in the LHN. Here are images from staging and production where I deleted the top message in a thread:
![2023-07-24_09-31-50](https://github.com/Expensify/App/assets/32969087/13bb541b-9537-4eda-9d1b-4529ef3d5bf8)

![2023-07-24_09-32-11](https://github.com/Expensify/App/assets/32969087/c9316c50-7de2-454b-8a2c-2306cfcbb591)

**Comment by cead22** [comment]
> * the parent report should have the old title

What is "the old title" here?

**Comment by rojiphil** [comment]
> * the (child) report title should now be [Deleted message] and have the last message sent as subtitle

@grgia Exactly, that's the issue here at #23445

**Comment by rojiphil** [comment]
> * the parent report should have the old title and the last message sent as subtitle

@grgia The above comment of yours is referring to this issue.

**Comment by cead22** [comment]
@grgia @rojiphil would you mind explaining the issue in detail with different words, I still don't understand what we're talking about

**Comment by grgia** [comment]
@cead22 
> What is "the old title" here?

Depends on the chat type, but for example:  DMs this would be the person you are talking to and for threads the Parent Report Action Message

**Comment by marcaaron** [comment]
the bug:

![2023-07-24_09-52-09](https://github.com/Expensify/App/assets/32969087/d2df443d-0693-41ee-9526-180b1327bbff)

**Comment by marcaaron** [comment]
Is maybe this logic just wrong?

https://github.com/Expensify/App/blob/c3fd1ea89fecb01477261565cb0acdfef325e797/src/libs/actions/Report.js#L900

If we have a `childVisibleActionCount` greater than `0` then we should show the last visible message... no?

**Comment by marcaaron** [comment]
Ok so here's where I'm at with this one...

1. The logic I linked above **definitely** looks wrong to me.
2. After changing that the report title doesn't update correctly because the LHN isn't re-rendering because we don't actually trigger any sort of update on the thread report itself when deleting the comment and only do it on the "original reportID" here:

https://github.com/Expensify/App/blob/a906877fed01ddfc32066493ba176c0863b437ab/src/libs/actions/Report.js#L872

I debugged the "optimistic data" we send when deleting a comment and there is no change on the thread report itself. If you trigger a render by tapping on another report row then the correct report name will show instantly.

**Comment by marcaaron** [comment]
Thanks, I'm focusing on unblocking the deploy. Is that other issue happening on production?

**Comment by rojiphil** [comment]
@marcaaron 
My [proposal here](https://github.com/Expensify/App/issues/23445#issuecomment-1648347158) resolves the issue regarding report name updation in thread report.
However, regarding [Deleted Message] as last message (subtitle), I am still not clear. Am investigating on this.

**Comment by marcaaron** [comment]
>it deleted the message rather than showing [DELETED MESSAGE]

Also this seems way less bad IMO...

**Comment by marcaaron** [comment]
>However, regarding [Deleted Message] as last message (subtitle), I am still not clear. Am investigating on this.

Solved [here](https://github.com/Expensify/App/issues/23401#issuecomment-1648566972)

@rojiphil I see you came to a similar conclusion about the thread report not updating.

I'm not sure about the solution though. I kind of works but is not really great because the report "last read" is something that also will be updated in the server when the report is opened and we could just as easily manipulate any other field on the report.

**Comment by marcaaron** [comment]
Would prefer to just do something like:

```js
    if (reportID !== originalReportID) {
        optimisticData.push({
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
            value: {isParentReportActionDeleted: true},
        });
    }
```

**Comment by marcaaron** [comment]
Reassigning C+ @allroundexperts

**Comment by s-alves10** [comment]
@marcaaron 

I don't think the current PR is the solution for this. Sure it works, but https://github.com/Expensify/App/issues/19722 occurs again because of this line
https://github.com/Expensify/App/blob/205c91c2cbe97e4cbc7fb9d004c4a9eb917ef922/src/libs/actions/Report.js#L900

I think we have find solution for both of these issues. Please check my [proposal](https://github.com/Expensify/App/issues/23401#issuecomment-1648305048) again

**Comment by s-alves10** [comment]
The introduction of `isLastMessageDeletedParentAction` has a lot of issues and I think we should remove it completely
If this is urgent, I can make a PR in an hour or 2

**Comment by marcaaron** [comment]
I think it will get us far enough to unblock the deploy. I'm going to leave a comment on that issue that the issue we are fixing there is not solved and/or needs some other solution so you can bring your proposal and any ideas there @s-alves10. Sound good?

**Comment by s-alves10** [comment]
Okay.

**Comment by marcaaron** [comment]
Works on staging. Left a comment [here](https://github.com/Expensify/App/issues/19722#issuecomment-1648818264).

**Comment by marcaaron** [comment]
Not overdue regression period is still ongoing

**Comment by marcaaron** [comment]
Or maybe someone needs to be paid? I have no idea 😄

**Comment by rojiphil** [comment]
@marcaaron  May I be considered for compensation as my [proposal/inputs](https://github.com/Expensify/App/issues/23445#issuecomment-1648347158) were also considered [here](https://github.com/Expensify/App/issues/23401#issuecomment-1648660828) and [here](https://github.com/Expensify/App/issues/23401#issuecomment-1648671903)?

**Comment by marcaaron** [comment]
@rojiphil you need to ask a BZ team member sorry I don't really handle individual payment requests. Refer to the guide here please https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#payment-for-contributions

**Comment by rojiphil** [comment]
Thanks @marcaaron for your comments. I am still getting exposed to the processes here as I am a newbie (just 3 months) with Expensify.

@cead22  As per [this comment in contributing guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#payment-for-contributions:~:text=We%20understand%20there%20may%20be%20cases%20where%20a%20selected%20proposal%20may%20take%20inspiration%20from%20a%20previous%20proposal.%20Unfortunately%2C%20it%E2%80%99s%20not%20possible%20for%20us%20to%20evaluate%20every%20individual%20case%20and%20we%20have%20no%20process%20that%20can%20efficiently%20do%20so.), may I be considered for compensation as my [proposal/inputs](https://github.com/Expensify/App/issues/23445#issuecomment-1648347158) were also considered [here](https://github.com/Expensify/App/issues/23401#issuecomment-1648660828) and [here](https://github.com/Expensify/App/issues/23401#issuecomment-1648671903)?

**Comment by marcaaron** [comment]
Adding `Bug` label so someone can help with [this](https://github.com/Expensify/App/issues/23401#issuecomment-1663266700) outstanding task.

**Comment by jliexpensify** [comment]
Ok, here is what I have so far:

- Bug Reporter: @gadhiyamanan  $250
- C+: @allroundexperts $1000 (request in New.Dot)

No Upworks job atm as it was Internal. Does that sound correct?

@allroundexperts can you complete the checklist before payment?

@cead22 can I get a ruling on this [comment](https://github.com/Expensify/App/issues/23401#issuecomment-1664895144)?

**Comment by JmillsExpensify** [comment]
Reviewed the details for @allroundexperts. This is approved for payment in NewDot.

**Comment by jliexpensify** [comment]
Bump @allroundexperts for the checklist (if applicable) and @cead22 for a ruling on this https://github.com/Expensify/App/issues/23401#issuecomment-1664895144, thanks!

**Comment by cead22** [comment]
TL;DR If it doesn't create a lot of extra work, I think we can compensate @rojiphil the bug reporter amount, as a sign of appreciation for the help with troubleshooting this.

Like @marcaaron, I've never made a decision about compensating anyone that wasn't specifically hired for a job, and my understanding about this issue that @marcaaron took over from me, is that the majority of the work was done by @marcaaron -- I believe he spent at least a couple of hours trying to figure this out, and we ultimately went with his solution.

That solution is similar to yours @rojiphil, but it's not clear to me that @marcaaron landed on that solution based on yours, because the timeline of events is not clear. 

@jliexpensify I believe this is ultimately your call, and if compensating @rojiphil the bug reporter amount is quick and easy, let's do it, otherwise, that's fine too as our guidelines are explicit about guaranteeing compensation for those that propose solutions and get hired for a job

**Comment by jliexpensify** [comment]
Thanks Carlos, appreciate it! I wasn't sure if you had any specific input as you were tagged, but I've included @rojiphil above for $250. Going to create an Upworks job to get everyone sorted.

**Comment by rojiphil** [comment]
Thanks @cead, @jliexpensify for the special bonus as a sign of appreciation. The process-oriented approach and transparent execution in getting work done through external contributors is what I really like about Expensify. Hope to contribute more with time.

**Comment by jliexpensify** [comment]
Everyone paid and job closed.
